mail_composer_cc_bcc/models/mail_template.py

```diff
@@ -60,142 +47,47 @@ def _generate_template_recipients(  # noqa:C901
+                    emails = tools.email_split(record_values.pop(field, ""))
+                    all_emails[field.split("_")[1]] += emails
+                    record_company = res_id_to_company.get(res_id)
+                    for mail in emails:
+                        email_to_res_ids[field.split("_")[1]].setdefault(
+                            mail, []
+                        ).append(res_id)
+                        if record_company:
+                            email_to_company[mail] = record_company
+
+            all_mails = all_emails["bcc"] + all_emails["cc"]
+            if all_mails:
                 customers_information = ModelSudo.browse(
                     res_ids
                 )._get_customer_information()
                 partners = self.env["res.partner"]._find_or_create_from_emails(
-                    all_emails,
+                    all_mails,
                     additional_values={
                         email: {
                             "company_id": email_to_company.get(email),
                             **customers_information.get(email, {}),
                         }
-                        for email in itertools.chain(all_emails, [False])
+                        for email in itertools.chain(all_mails, [False])
                     },
                 )
-                for original_email, partner in zip(all_emails, partners):  # noqa: B905
+                for original_email, partner in zip(all_mails, partners, strict=False):
                     if not partner:
                         continue
-                    for res_id in email_to_res_ids[original_email]:
-                        render_results[res_id].setdefault("partner_ids", []).append(
-                            partner.id
-                        )
-
-        # update 'partner_to' rendered value to 'partner_ids'
-        all_partner_to = {
-            pid
-            for record_values in render_results.values()
-            for pid in self._parse_partner_to(record_values.get("partner_to", ""))
-        }
-        existing_pids = set()
-        if all_partner_to:
-            existing_pids = set(
-                self.env["res.partner"].sudo().browse(list(all_partner_to)).exists().ids
-            )
-        for res_id, record_values in render_results.items():  # noqa: B007
-            partner_to = record_values.pop("partner_to", "")
-            if partner_to:
-                tpl_partner_ids = (
-                    set(self._parse_partner_to(partner_to)) & existing_pids
-                )
-                record_values.setdefault("partner_ids", []).extend(tpl_partner_ids)
+                    for field in ["bcc", "cc"]:
+                        if original_email in email_to_res_ids[field]:
+                            for res_id in email_to_res_ids[field][original_email]:
+                                res[res_id].setdefault(
+                                    f"partner_{field}_ids", []
```

Comment:
returning the value of `partner_bcc_ids` and `partner_cc_ids` here is dangerous. We can not trust on the context ` context.get("default_composition_mode")`. 
The most important thing is that, all the key in returned value must be in the `render_fields`.
So, don't compute  `partner_bcc_ids` and `partner_cc_ids`  here. But, do this in `def _compute_partner_cc_bcc_ids(self):` of `mail.compose.message`.
```
                if rendered_values.get("email_cc"):
                    partners = self.env["res.partner"]._find_or_create_from_emails(.... from rendered_values.get("email_cc")....)
                    if partners:
                        partner_cc_ids = partners.ids
               ........ same for email_bcc ............................
```

Reply:
Hi anh @nguyenminhchien, indeed, "context" looks like not a good approach

I just have an idea, how does it sound to you ?
- As we decided to not interfere with mass_mailing, in `https://github.com/OCA/social/pull/1203`
- So, patch `_prepare_mail_values` to filter out add-on fields

Pros
- Keep consistency with Odoo's flow: recipients are generated in `_generate_template_recipients`
- Processing those mails once. If doing that in `_compute_partner_cc_bcc_ids`, changing any dependency results in re-trigger.

Cons
- Yes, quite resource-consuming (adding cc/bcc mail back and forth). But is it common to have cc/bcc in context of mass mailing :thinking:.

Reply:
Hi Trí,
You should not return the value of any field which is not in `render_fields` (method `def _generate_template_recipients()`). Particularly,  `partner_cc_ids` and `partner_bcc_ids` are not in `render_fields`. So, you should not include them in the returned value.
![image](https://github.com/user-attachments/assets/fc03c2ba-aa5e-4caf-8594-7aa7176bf86e)

Reply:
I think `def _compute_partner_cc_bcc_ids` is the best suited-place where to handle computing the value of `partner_cc_ids` and `partner_bcc_ids`.